### PR TITLE
NetworkPkg: Improve GetBootFile() code flow

### PR DESCRIPTION
**Pull request by SaloniKasbekar**
Introduce state machine to improve the code flow in GetBootFile() to make it more readable. Allows new states to be easily added without adding further nested ifs.

# Description
   
Introduce state machine to improve the code flow in GetBootFile()
to make it more readable. Allows new states to be easily added
without adding further nested ifs.

- [ ] Breaking change?

- [ ] Impacts security?

- [ ] Includes tests?

## How This Was Tested

CI

## Integration Instructions

N/A

### CODE REVIEW

File: NetworkPkg/HttpBootDxe/HttpBootImpl.c [hunk 1]
```diff
@@ -274,6 +274,119 @@ HttpBootDhcp (
...
+                                   the request.
+  @retval EFI_ACCESS_DENIED        Server authentication failed.
+  @retval Others                   Unexpected error happened.
+**/
+EFI_STATUS
+HttpBootGetBootFileCaller (
+  IN     HTTP_BOOT_PRIVATE_DATA  *Private,
+  IN OUT UINTN                   *BufferSize,
+  IN     VOID                    *Buffer        OPTIONAL,
+  OUT HTTP_BOOT_IMAGE_TYPE       *ImageType
```
Comment (Zclarkwilliams) [inline NetworkPkg/HttpBootDxe/HttpBootImpl.c:302]: align HTTP_BOOT_IMAGE_TYPE with other parameters

Comment (SaloniKasbekar) [reply NetworkPkg/HttpBootDxe/HttpBootImpl.c]: Done

**Review by Zclarkwilliams — APPROVED** [review]

**Review by mdkinney — APPROVED** [review]

**Comment by mdkinney** [comment]
@SaloniKasbekar Is this change required for edk2-stable202408?

**Comment by SaloniKasbekar** [comment]
> @SaloniKasbekar Is this change required for edk2-stable202408?

Not needed. Can wait till after the stable tag.

**Pull request merged**